File: core/src/main/java/y2k/joyreactor/presenters/CreateCommentPresenter.java
package y2k.joyreactor.presenters;

import y2k.joyreactor.Profile;
import y2k.joyreactor.platform.Navigation;
import y2k.joyreactor.services.CommentService;
import y2k.joyreactor.services.ProfileService;

/**
 * Created by y2k on 10/4/15.
 */
public class CreateCommentPresenter {

    private View view;
    private CommentService service;

    public CreateCommentPresenter(View view, ProfileService profileService, CommentService service) {
        this.view = view;
        this.service = service;

        profileService.get().subscribe(view::setUser, Throwable::printStackTrace);
    }

    public void create(String commentText) {
        view.setIsBusy(true);
        service.createComment("2219757", commentText)
                .subscribe(_void -> {
                    Navigation.getInstance().closeCreateComment();
                    view.setIsBusy(false);
                }, Throwable::printStackTrace);
    }

    public interface View {

        void setIsBusy(boolean isBusy);

        void setUser(Profile profile);
    }
}

File: core/src/main/kotlin/y2k/joyreactor/presenters/CreateCommentPresenter.kt
package y2k.joyreactor.presenters

import y2k.joyreactor.model.Profile
import y2k.joyreactor.common.subscribeOnMain
import y2k.joyreactor.platform.NavigationService
import y2k.joyreactor.services.CommentService
import y2k.joyreactor.services.ProfileService

/**
 * Created by y2k on 10/4/15.
 */
class CreateCommentPresenter(
    private val view: CreateCommentPresenter.View,
    private val profileService: ProfileService,
    private val service: CommentService) {

    init {
        profileService
            .getProfile()
            .subscribeOnMain { view.setUser(it) }
    }

    fun create(commentText: String) {
        view.setIsBusy(true)
        service
            .createComment("2219757", commentText)
            .subscribeOnMain {
                NavigationService.instance.closeCreateComment()
                view.setIsBusy(false)
            }
    }

    interface View {

        fun setIsBusy(isBusy: Boolean)

        fun setUser(profile: Profile)
    }
}